system-test/test-middleware.ts

```diff
@@ -0,0 +1,44 @@
+/*!
+ * Copyright 2018 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/* AN EXAMPLE RATHER THAN A TEST AT THIS POINT */
+
+import {middleware as lb} from '@google-cloud/logging-bunyan';
+import * as express from 'express';
+
+const app = express();
+
+const {logger, mw} = lb.express({
+  projectId: 'squirrely-squirrel',
+  keyFilename: '/path/to/keyfile',
+  level: 'info'
+});
+
+app.use(mw);  // this should probably be the first middleware you mount.
+
+app.get('/', (req, res) => {
+  // req.log gives you a bunyan-like logger API. These logs gets correlated with
+  // the corresponding request – so they will be bundled with the corresponding
+  // request log.
+  req.log.info('hello');
```

Comment:
I'm a little curious - how does the TypeScript typing work out here?  I don't entirely understand how `req.log` compiles tbh.

Reply:
We do/should export `AnnotatedRequest` (or maybe a different name) that users can use.